src/Umbraco.Core/Persistence/Repositories/Implement/DocumentRepository.cs

```diff
@@ -683,16 +695,30 @@ public void AddOrUpdatePermissions(ContentPermissionSet permission)
 
         /// <inheritdoc />
         public override IEnumerable<IContent> GetPage(IQuery<IContent> query,
-            long pageIndex, int pageSize, out long totalRecords,            
+            long pageIndex, int pageSize, out long totalRecords,
             IQuery<IContent> filter, Ordering ordering)
         {
             Sql<ISqlContext> filterSql = null;
 
+            var temp = Query<IContent>().Where(x => x.Name.Contains("foo"));
```

Comment:
I have no idea what this is doing by reading this code. I think this needs a bit of documentation as to what this is all doing and what it is achieving.